instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -48,6 +50,10 @@ public OpenTelemetry openTelemetry(
         tracerProviderBuilder
             .setSampler(Sampler.traceIdRatioBased(samplerProperties.getProbability()))
             .build();
-    return OpenTelemetrySdk.builder().setTracerProvider(tracerProvider).buildAndRegisterGlobal();
+
+    return OpenTelemetrySdk.builder()
+        .setTracerProvider(tracerProvider)
+        .setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
```

Comment:
Can you make it configurable? You could make it similar to how exporters are currently configured (some propagators, like B3, are not included in the API and are present in the `opentelemetry-extension-trace-propagators` jar)

Reply:
I'll try my best. To to make sure I got that correctly, this would mean creating AutoConfigurations for different kinds of propagators (as seen in case of exporters). The configuration should then be passed as an argument. Correct me if I'm wrong :)

Reply:
Hmm, after thinking about it for a bit my previous idea is a bit lacking: opposite to span exporters, it makes sense to use multiple propagators (the detaults should be `w3c` and `baggage` after all), so implementing this exactly the same as exporters and using things like `@ConditionalOnMissingBean` (because there can be more than one) or `@ConditionalOnClass` (because once you add the propagators extension all of them will be on classpath) is far from ideal.

How about implementing this similarly to spring-sleuth-otel?
In particular see https://github.com/spring-cloud-incubator/spring-cloud-sleuth-otel/blob/main/spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelPropagationConfiguration.java#L97 and https://github.com/spring-cloud-incubator/spring-cloud-sleuth-otel/blob/main/spring-cloud-sleuth-otel/src/main/java/org/springframework/cloud/sleuth/otel/propagation/CompositeTextMapPropagator.java#L50. 
You could use a single property: a list of propagator names (with default value `baggage,w3c`) and build a composite one depending on what's on that list. WDYT?

Reply:
Makes sense, I'll try implementing it that way. So both `w3c` and `baggage` should be added regardless of the value of the property?

Reply:
No, they should be used as the default value - if user doesn't specify any propagator at all these two should be set.